File: question/neetcode150/array_and_hashing/encode-and-decode-strings.py
from typing import List

class Solution:

    def encode(self, ss: List[str]) -> str:
        a = ""
        for s in ss:
            a += str(len(s)) + "#" + s

        return a

    def decode(self, ss: str) -> List[str]:
        a = []

        i = 0
        while i < len(ss):
            n = ""
            while ss[i] != "#":
                n+=ss[i]
                i+=1
            i+=1
            a.append(ss[i:i+int(n)])
            i+=int(n)

        return a

File: question/neetcode150/array_and_hashing/encode-and-decode-strings.cpp
#include <bits/stdc++.h>
 
using namespace std;

class Solution {
public:

    string encode(vector<string>& ss) {
        string a;

        for(auto s: ss){
            a += to_string(s.size()) + '#' + s;
        }

        return a;
    }

    vector<string> decode(string s) {
        vector<string> a;

        int i = 0;
        while(i<s.size()){
            int j = 0;
            while (s[i+j]!='#'){
                j++;
            }

            int n = stoi(s.substr(i,j));
            a.push_back(s.substr(i+j+1, n));
            i+=j+1+n;
        }

        return a;
    }
};
